contracts/JBTieredLimitedNFTRewardDataSource.sol

```diff
@@ -312,35 +321,98 @@ contract JBTieredLimitedNFTRewardDataSource is
     Mints a token for a given contribution to the beneficiary.
 
     @dev
-    `_data.metadata` should include the number of rewards being requested in bits 32-39.
+    `_data.metadata` should include the reward tiers being requested in increments of 8 bits starting at bits 32.
 
     @param _data The Juicebox standard project contribution data.
   */
   function _processContribution(JBDidPayData calldata _data) internal override {
-    // Make the contribution is being made in the expected token.
+    // Make sure the contribution is being made in the expected token.
     if (_data.amount.token != contributionToken) return;
 
-    // Keep a reference to the number of rewards being processed. Skip the first 32 bits which are used by the JB protocol.
-    uint256 _numRewards = uint256(uint8(_bytesToUint(_data.metadata) << 32));
+    // Check if the metadata is filled with the needed information
+    if (_data.metadata.length > 32) {
+      // Decode the metadata, Skip the first 32 bits which are used by the JB protocol.
+      (, uint8[] memory _decodedMetadata) = abi.decode(_data.metadata, (bytes32, uint8[]));
 
-    // Keep a reference to the value remaining.
-    uint256 _remainingValue = _data.amount.value;
+      // Mint rewards if they were specified. If there are no rewards but a default NFT should be minted, do so.
+      if (_decodedMetadata.length != 0)
+        return _mintAll(_data.amount.value, _decodedMetadata, _data.beneficiary);
+    }
 
-    // Keep a reference to the tier ID being iterated on.
-    uint256 _tierId;
+    // Fallback for if we were not able to mint using the metadata
+    if (shouldMintByDefault) return _mintBestAvailableTier(_data.amount.value, _data.beneficiary);
+  }
+
+  /** 
+    @notice
+    Mints a token in the best available tier.
+
+    @param _amount The amount to base the mint on.
+    @param _beneficiary The address to mint for.
+  */
+  function _mintBestAvailableTier(uint256 _amount, address _beneficiary) internal {
+    // Keep a reference to the number of tiers.
+    uint256 _numTiers = numTiers;
+
+    // Keep a reference to the tier being iterated on.
+    JBNFTRewardTier storage _tier;
+
+    // Loop through each tier. Go from most valuable tier to least valuable.
+    for (uint256 _i = _numTiers; _i != 0; ) {
+      // Set the tier being iterated on. Tier's are 1 indexed.
+      _tier = tiers[_i];
+
+      // Mint if the contribution value is at least as much as the floor, there's sufficient supply.
+      if (_tier.contributionFloor <= _amount && _tier.remainingQuantity != 0) {
+        // Mint the tokens.
+        uint256 _tokenId = _mintForTier(_i, _tier, _beneficiary);
+
+        emit Mint(_tokenId, _i, _beneficiary, _amount, 0, msg.sender);
```

Comment:
Quantity should still be 1 (instead of 0)?

Reply:
0 gives us some insight that this was triggered without specifying a reward tier